climakitae/core/data_interface.py

```diff
@@ -262,28 +264,40 @@ def _get_subarea_from_shape_index(boundary_dataset, shape_index):
             crs="EPSG:4326",
         )
     elif area_subset != "none":
-        shape_index = int(_geography_choose[area_subset][cached_area])
+        # `if-condition` added for catching errors with delays in rendering cached area.
```

Comment:
This is something I implemented to catch errors because of delays that were occurring when re-rendering the panel (sometimes `cached_area` would return None, but that would cause defining `shape_indicies` to fail). This error would occur when subsetting the data would be selected by none. I would love to get another pair of eyes to see if this is a good fix or if there is something more concrete I could implement instead.

Reply:
cool